doc/source/data/loading-data.rst

```diff
@@ -1,607 +1,531 @@
-      instance to :func:`read_parquet() <ray.data.read_parquet>`.
+            import ray
 
-      .. literalinclude:: ./doc_code/loading_data_untested.py
-        :language: python
-        :start-after: __read_parquet_hdfs_with_fs_begin__
-        :end-before: __read_parquet_hdfs_with_fs_end__
+            ds = ray.data.read_csv("s3://anonymous@ray-example-data/iris.csv")
 
-    .. tab-item:: GCS
+            print(ds.schema())
 
-      Data can be read from Google Cloud Storage by providing a configured
-      `gcsfs GCSFileSystem <https://gcsfs.readthedocs.io/en/latest/>`__, where the
-      appropriate Google Cloud project and credentials can be specified.
+        .. testoutput::
 
-      .. note::
-        This example is not runnable as-is; you'll need to point it at your GCS bucket and
-        configure your GCP project and credentials.
+            Column             Type
+            ------             ----
+            sepal length (cm)  double
+            sepal width (cm)   double
+            petal length (cm)  double
+            petal width (cm)   double
+            target             int64
 
-      .. literalinclude:: ./doc_code/loading_data_untested.py
-        :language: python
-        :start-after: __read_parquet_gcs_begin__
-        :end-before: __read_parquet_gcs_end__
+        To read other file formats, see the :ref:`Input/Output reference <input-output>`.
 
-      .. tip::
-        To verify that your GCP project and credentials are set up, validate
-        that the GCS `filesystem` has permissions to read the input `path`.
+Handling compressed files
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-        .. literalinclude:: ./doc_code/loading_data_untested.py
-          :language: python
-          :start-after: __validate_parquet_gcs_begin__
-          :end-before: __validate_parquet_gcs_end__
+To read a compressed file, specify ``compression`` in ``arrow_open_stream_args``.
+You can use any `Codec supported by Arrow <https://arrow.apache.org/docs/python/generated/pyarrow.CompressedInputStream.html>`__.
 
-        For more examples, see the `GCSFS Documentation <https://gcsfs.readthedocs.io/en/latest/#examples>`__.
+.. testcode::
 
-    .. tab-item:: ADL/ABS (Azure)
+    import ray
 
-      Data can be read from Azure Blob Storage by providing a configured
-      `adlfs AzureBlobFileSystem <https://github.com/fsspec/adlfs>`__, where the appropriate
-      account name and account key can be specified.
+    ds = ray.data.read_csv(
+        "s3://anonymous@ray-example-data/iris.csv.gz",
+        arrow_open_stream_args={"compression": "gzip"},
+    )
 
-      .. literalinclude:: ./doc_code/loading_data_untested.py
-        :language: python
-        :start-after: __read_parquet_az_begin__
-        :end-before: __read_parquet_az_end__
+Loading in-memory data
+======================
 
-Reading from Local Storage
-==========================
+Loading data from common libraries
```

Comment:
thinking about it more, I'm not sure if users would really understand what "common libraries" means.

should we rename this "Loading from other Python libraries"